### PR TITLE
[kube-prometheus-stack] Add uid to grafana default datasources

### PR DESCRIPTION
**Pull request by Keramblock**
Signed-off-by: Michael Kirpichev <m.kirpichev@haut.ai>
#### What this PR does / why we need it:
Today I tried to migrate to new grafana alerting and noticed following diff in one of my dashboards:
Screenshot 2022-02-21 at 15 17 08
So now grafana relates on uid to reference data source from dashboard. To avoid any potential issues I think that is a good idea to fix uid of default data sources. I dont think this change will affect anybody with old alerts and will ease migration to new system.

P.S. 
Grafana dashboard migration results after uid was specified:
изображение

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

File: charts/kube-prometheus-stack/templates/grafana/configmaps-datasources.yaml [hunk 1]
```diff
@@ -24,6 +24,7 @@ data:
 {{- if .Values.grafana.sidecar.datasources.defaultDatasourceEnabled }}
     - name: Prometheus
       type: prometheus
+      uid: promeheus
```
Comment (hlomzik) [inline charts/kube-prometheus-stack/templates/grafana/configmaps-datasources.yaml:27]: I guess it should be "prometheus" with "t" missing. And also in the second uid below

Comment (Keramblock) [reply charts/kube-prometheus-stack/templates/grafana/configmaps-datasources.yaml]: Thx! gixed

**Comment by TeddyAndrieux** [comment]
Good I was about to open a PR to do this :+1: 

If I may, @Keramblock could you make those UID configurable please see #1720 (otherwise I can open another PR to do this)

**Comment by Keramblock** [comment]
> Good I was about to open a PR to do this 👍
> 
> If I may, @Keramblock could you make those UID configurable please see #1720 (otherwise I can open another PR to do this)

I am not sure if this is good idea to make only UID configurable in that PR, because:

1. It is defaults and we can make our own, if we need something else
2. Other fields of it are not parametrised
3. In guidlines there is line that we need to create PR that solves only one problem at a time, so we need another one for that problem
4. I am not sure if maintainers will approve such change

**Comment by TeddyAndrieux** [comment]
> I am not sure if this is good idea to make only UID configurable in that PR, because:

> 1. It is defaults and we can make our own, if we need something else

I agree we can, kind of, work around it, but we may want to just override the default UID so that we don't need to "know" the content of the Datasource config file when we apply an helm chart

> 2. Other fields of datasource are not parametrised

Agree but we need to start somewhere :smiley: and I think this field makes sense since we can use it in dashboard

> 3. In guidlines there is line that we need to create PR that solves only one problem at a time, so we need another one for that problem

Fine, I can do another PR for it, don't worry was just asking since it related but never mind

> 4. I am not sure if maintainers will approve such change

I think/hope they will :)

**Comment by TeddyAndrieux** [comment]
> I am not sure if this is good idea to make only UID configurable in that PR, because:

> 1. It is defaults and we can make our own, if we need something else

I agree we can, kind of, work around it, but we may want to just override the default UID so that we don't need to "know" the content of the Datasource config file when we apply an helm chart

> 2. Other fields of datasource are not parametrised

Agree but we need to start somewhere :smiley: and I think this field makes sense since we can use it in dashboard

> 3. In guidlines there is line that we need to create PR that solves only one problem at a time, so we need another one for that problem

Fine, I can do another PR for it, don't worry was just asking since it related but never mind

> 4. I am not sure if maintainers will approve such change

I think/hope they will :)

**Review by Xtigyro — DISMISSED** [review]

**Review by Xtigyro — APPROVED** [review]

**Comment by Xtigyro** [comment]
Thanks for the contribution, @Keramblock! 🥇 

LGTM.

**Pull request merged**